trieste/models/keras/models.py

```diff
@@ -332,3 +338,176 @@ def optimize(self, dataset: Dataset) -> None:
 
         x, y = self.prepare_dataset(dataset)
         self.model.fit(x=x, y=y, **self.optimizer.fit_args)
+
+
+class MonteCarloDropout(KerasPredictor, TrainableProbabilisticModel):
+    """
+    A :class:`~trieste.model.TrainableProbabilisticModel` wrapper for Monte Carlo dropout
+    built using Keras.
+
+    Monte Carlo dropout is a sampling method for approximate Bayesian computation, mathematically
+    equivalent to an approximation to a probabilistic deep Gaussian Process <cite data-cite="gal2015simple"/>
+    in the sense of minimizing the Kullback-Leibler divergence between an approximate distribution 
+    and the posterior of a deep GP. This model is attractive due to its simplicity, as it amounts 
+    to a re-tooling of the dropout layers of a neural network to also be active during testing, 
+    and performing several forward passes through the network with the same input data. The 
+    resulting distribution of the outputs of the different passes are then used to estimate the
+    first two moments of the predictive distribution. Note that increasing the number of passes
+    increases accuracy at the cost of a higher computational burden.
+
+    The uncertainty estimations of the original paper have been subject to extensive scrutiny, and
+    it has been pointed out that the quality of the uncertainty estimates is tied to parameter
+    choices which need to be calibrated to accurately account for model uncertainty. A more robust
+    alternative is MC-DropConnect, an approach that generalizes the prior idea by applying dropout
+    not to the layer outputs but directly to each weight (see <cite data-cite="mobiny2019"/>). 
+
+    We provide classes for constructing neural networks with Monte Carlo dropout using Keras
+    (:class:`~trieste.models.keras.DropoutNetwork`) in the `architectures` package that should be
+    used with the :class:`~trieste.models.keras.MonteCarloDropout` wrapper. There we also provide
+    an application of MC-DropConnect, by setting the argument `dropout` to 'dropconnect'.
+
+    Note that currently we do not support setting up the model with dictionary configs and saving
+    the model during Bayesian optimization loop (``track_state`` argument in
+    :meth:`~trieste.bayesian_optimizer.BayesianOptimizer.optimize` method should be set to `False`).
+    """
+    def __init__(
+        self,
+        model: DropoutNetwork,
+        optimizer: Optional[KerasOptimizer] = None,
+        num_passes: int = 100,
+        learning_rate: float = 0.01
```

Comment:
what is this input? its not declared below as self?

Reply:
It is already deleted in other branches, but not pushed. It is a residual argument from when Clinton and I discussed how to be go about resetting LR.